modules/landing-pages/module.php

```diff
@@ -212,53 +269,169 @@ private function register_landing_page_cpt() {
+	 *
+	 * In some cases (such as when a site uses custom permalink structures), WordPress's WP_Query does not identify a
+	 * Landing Page's URL as a post belonging to the Landing Page CPT. Some cases are handled successfully by the
+	 * adjust_landing_page_query() method, but some are not and still trigger a 404 process. This method handles such
+	 * cases by overriding the $wp_query global to fetch the correct landing page post entry.
+	 *
+	 * For example, since Landing Pages slugs come directly after the site domain name, WP_Query might parse the post
+	 * as a category page. Since there is no category matching the slug, it triggers a 404 process. In this case, we
+	 * run a query for a Landing Page post with the passed slug ($query->query['category_name']. If a Landing Page
+	 * with the passed slug is found, we override the global $wp_query with the new, correct query.
+	 *
+	 * @param $current_value
+	 * @param $query
+	 * @return false
+	 */
+	private function handle_404( $current_value, $query ) {
+		global $wp_query;
+
+		// If another plugin/theme already used this filter, exit here to avoid conflicts.
+		if ( $current_value ) {
+			return $current_value;
+		}
+
+		if (
+			// Make sure we only intervene in the main query.
+			! $query->is_main_query()
+			// If a post was found, this is not a 404 case, so do not intervene.
+			|| ! empty( $query->posts )
+			// This filter is only meant to deal with wrong queries where the only query var is 'category_name'.
+			// If there is no 'category_name' query var, do not intervene.
+			|| empty( $query->query['category_name'] )
+			// If the query is for a real taxonomy (determined by it including a table to search in, such as the
+			// wp_term_relationships table), do not intervene.
+			|| ! empty( $query->tax_query->table_aliases )
+		) {
+			return false;
+		}
+
+		// Search for a Landing Page with the same name passed as the 'category name'.
+		$possible_new_query = new \WP_Query( [
+			'post_type' => self::CPT,
+			'name' => $query->query['category_name'],
+		] );
+
+		// Only if such a Landing Page is found, override the query to fetch the correct page.
+		if ( ! empty( $possible_new_query->posts ) ) {
+			$wp_query = $possible_new_query; //phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		}
+
+		return false;
+	}
+
 	public function __construct() {
 		$this->permalink_structure = get_option( 'permalink_structure' );
 
 		$this->register_landing_page_cpt();
 
-		// Landing Pages should act like pages, including in their permalink structure. If the permalink structure is
-		// `/%postname%/`, the following hooks change the permalink to remove the CPT slug from it.
-		if ( ! is_admin() && false !== strpos( $this->permalink_structure, '/%postname%/' ) ) {
+		// If there is a permalink structure set to the site, run the hooks that modify the Landing Pages permalinks to
+		// match WordPress' native 'Pages' post type.
+		if ( '' !== $this->permalink_structure ) {
```

Comment:
Why should it run if `is_admin()`?

Reply:
preview?

Reply:
added an `is_admin` or `is_ajax` check for the `pre_get_posts` and `pre_handle_404` filters